Content.Shared/Damage/Components/RequireProjectileTargetComponent.cs

```diff
@@ -0,0 +1,11 @@
+namespace Content.Shared.Damage.Components;
+
+/// <summary>
+/// Prevent the object from getting hit by projetiles unless you target the object.
+/// </summary>
+[RegisterComponent]
```

Comment:
networked component, access and also network Active with AutoNetworkedField attribute and dirty when you update it (add a `SetActive(Entity<COMPONENT> ent, bool value) { if (ent.Comp.Active == value) return; ent.Comp.Active = value; Dirty(ent)}` method)